pkg/messaging/nats/pubsub.go

```diff
@@ -77,7 +78,16 @@ func (ps *pubsub) Subscribe(id, topic string, handler messaging.MessageHandler)
 	case true:
 		// Check topic ID
 		if _, ok := s[id]; ok {
-			return ErrAlreadySubscribed
+			// Unlocking, so that Unsubscribe() can access ps.subscriptions
+			ps.mu.Unlock()
+			if err := ps.Unsubscribe(id, topic); err != nil {
+				return err
+			}
+			ps.mu.Lock()
+			if len(s) == 0 {
```

Comment:
This should work, but only because maps are passed around by pointer (so that `s` is updated by the previous Unsubscribe call), not because it's a good design. Test it.